include/triton/Dialect/TritonGPU/IR/Dialect.h

```diff
@@ -35,7 +35,9 @@ SmallVector<unsigned> getContigPerThread(Attribute layout);
 
 SmallVector<unsigned> getThreadsPerCTA(const Attribute &layout);
 
-SmallVector<unsigned> getShapePerCTA(const Attribute &layout);
+SmallVector<unsigned>
+getShapePerCTA(const Attribute &layout,
+               ArrayRef<int64_t> blockShape = ArrayRef<int64_t>());
```

Comment:
- If this is only for MMAv1, I prefer to add a separate getShapePerCTAMMAv1(). Or else we should well define  the meaning of "blockShape".

- Plus that, I don't understand the logic here. ShapePerCTA should only be decided by MMAEncodingAttr(v1). I guess this is also part of the TODO "mmaV1 mma block could be integrated into the existing distributed layout", is it?

Reply:
For mmav1, the shapePerCTA is the shape of the tensor, which is quite different from mmav2.
Based on the current implementation, it needs the recursive call getShapePerCTA from the parent layout of slice and dotOperand layout, which needs to be shared.

I've renamed the argument name from blockShape to tensorShape for clarity.